src/com/cloudogu/ces/cesbuildlib/Trivy.groovy

```diff
@@ -0,0 +1,133 @@
+package com.cloudogu.ces.cesbuildlib
+
+class Trivy implements Serializable {
+    static final String DEFAULT_TRIVY_VERSION = "0.57.1"
+    private script
+    private Docker docker
+    private String trivyVersion
+    private String trivyDirectory = "trivy"
+
+    Trivy(script, String trivyVersion = "0.57.1", Docker docker = new Docker(script)) {
+        this.script = script
+        this.trivyVersion = trivyVersion
+        this.docker = docker
+    }
+
+    /**
+     * Scans an image for vulnerabilities.
+     * Notes:
+     * - Use a .trivyignore file for allowed CVEs
+     * - This function will generate a JSON formatted report file which can be converted to other formats via saveFormattedTrivyReport()
+     * - Evaluate via exit codes: 0 = no vulnerability; 1 = vulnerabilities found; other = function call failed
+     *
+     * @param imageName The name of the image to be scanned; may include a version tag
+     * @param severityLevel The vulnerability level to scan. Can be a member of TrivySeverityLevel or a custom String (e.g. 'CRITICAL,LOW')
+     * @param strategy The strategy to follow after the scan. Should the build become unstable or failed? Or Should any vulnerability be ignored? (@see TrivyScanStrategy)
+     * @param additionalFlags Additional Trivy command flags
+     * @param trivyReportFile Location of Trivy report file. Should be set individually when scanning multiple images in the same pipeline
+     * @return Returns true if the scan was ok (no vulnerability found); returns false if any vulnerability was found
+     */
+    boolean scanImage(
+        String imageName,
+        String severityLevel = TrivySeverityLevel.CRITICAL,
+        String strategy = TrivyScanStrategy.UNSTABLE,
+        // Avoid rate limits of default Trivy database source
+        String additionalFlags = "--db-repository public.ecr.aws/aquasecurity/trivy-db --java-db-repository public.ecr.aws/aquasecurity/trivy-java-db",
+        String trivyReportFile = "trivy/trivyReport.json"
+    ) {
+        Integer exitCode = docker.image("aquasec/trivy:${trivyVersion}")
```

Comment:
We should not hardcode the trivy image as Docker Hub has a rate limiting, which makes trouble already. If we make this configurable we could use a mirror, like a artifact repo in our own cloud.

Reply:
Resolved by https://github.com/cloudogu/ces-build-lib/commit/174645f059efb50fda1f4606d296bf964992382a